Plugin/Sales/Model/Spi/CreditmemoResource.php

```diff
@@ -118,9 +129,39 @@ public function aroundSave(
             && $this->avaTaxConfig->getTaxMode($entity->getStoreId()) == Config::TAX_MODE_ESTIMATE_AND_SUBMIT
             && $this->avaTaxConfig->isAddressTaxable($address, $storeId)
         ) {
-
             // Add this entity to the avatax processing queue if this is a new entity
             if ($isObjectNew) {
+                $itemTaxes = $this->associatedTaxableRepository->getItemAssociatedTaxablesForOrder($order->getId());
```

Comment:
Move all of this code into its own method. Wrap with a try/catch block (with error logging).